subsys/bluetooth/host/l2cap.c

```diff
@@ -797,9 +795,7 @@ static void l2cap_chan_rx_give_credits(struct bt_l2cap_le_chan *chan,
 {
 	BT_DBG("chan %p credits %u", chan, credits);
 
-	while (credits--) {
-		k_sem_give(&chan->rx.credits);
-	}
+	atomic_sub(&chan->rx.credits, credits);
```

Comment:
Why are you doing `atomic_sub()` here and `atomic_add()` in `l2cap_chan_tx_give_credits()`. Shouldn't they both be `atomic_add()`?

Reply:
Will fix that, not sure how this worked for me but it obviously wrong.